xtask/src/main.rs

```diff
@@ -1,37 +1,28 @@
-use std::{
-    collections::{HashMap, HashSet},
-    fs,
-    path::Path,
-    process::Command,
-    str,
-    sync::Mutex,
-};
-
-use anyhow::{anyhow, Context, Result};
+use std::{fs, path::Path, process::Command, str, sync::Mutex};
+
+use anyhow::{anyhow, Context};
 use console::Style;
 use once_cell::sync::Lazy;
 use similar::{ChangeTag, TextDiff};
 use structopt::StructOpt;
 
+mod targets;
+
 static ALL_ERRORS: Lazy<Mutex<Vec<String>>> = Lazy::new(|| Mutex::new(vec![]));
 
 #[derive(Debug, StructOpt)]
 struct Options {
-    #[structopt(long, short, help = "keep target toolchains that were installed as dependency")]
-    keep_targets: bool,
-
-    #[structopt(
-        long,
-        short,
-        help = "treat compiler warnings as errors (RUSTFLAGS=\"--deny warnings\")"
-    )]
-    deny_warnings: bool,
-
     #[structopt(subcommand)]
     cmd: TestCommand,
+    #[structopt(long, short)]
+    /// Treat compiler warnings as errors (`RUSTFLAGS="--deny warnings"`)
```

Comment:
I prefer to place doc comments before any attributes
```suggestion
    /// Treat compiler warnings as errors (`RUSTFLAGS="--deny warnings"`)
    #[structopt(long, short)]
```